spacy/cli/debug_data.py

```diff
@@ -193,6 +194,66 @@ def debug_data(
     else:
         msg.info("No word vectors present in the package")
 
+    if "spancat" in factory_names:
+        model_labels = _get_labels_from_model(nlp, "spancat")
+        labels = [list(l.keys()) for l in gold_train_data["spancat"].values()]
+        has_low_data_warning = False
+        has_no_neg_warning = False
+        has_ws_ents_error = False
+
+        msg.divider("Span Categorization")
+        msg.info(f"{len(model_labels)} label(s)")
+
+        # Check if some model labels are not present in actual dataset
+        msg.text("Labels in train data: ")
+        for span_key, data_labels in gold_train_data["spancat"].items():
+            msg.text(
+                f"Key: {span_key}, {_format_labels(data_labels.items(), counts=True)}"
+            )
+
+        missing_labels = model_labels - set(itertools.chain(*labels))
+        if missing_labels:
+            msg.warn(
+                "Some model labels are not present in the train data. The "
+                "model performance may be degraded for these labels after "
+                f"training: {_format_labels(missing_labels)}."
+            )
```

Comment:
I think it would make sense to check this per `spans_key`.

Actually, I think for all these checks it would make sense to run all the checks per `spancat`+`spans_key` as provided in the config, since the main concern for `debug data` should be the spans that are being used for training in this configuration.

This makes the checks a bit different from `ner`, where there's only one possible place to store training data for an `ner` component.

Reply:
Ahh gotcha, will update this!

Reply:
Hmm, this might be a bit bigger than I imagined. I don't think the `Language.labels` attribute for SpanCat [differentiates between span keys](https://github.com/explosion/spaCy/blob/master/spacy/pipeline/spancat.py#L242) (cf. [how `add_label` is called in SpanCat](https://github.com/explosion/spaCy/blob/master/spacy/pipeline/spancat.py#L408)). When calling `pipe.cfg["labels"]` it will always return a "flattened" list irrespective of the span_key.

It is possible to change `pipe.cfg["labels"]` into a dictionary (i.e., `Dict[str, List[str]]` where key is `spans_key` and value is the corresponding labels) but it might affect/break a few things down the line. 

I can probably do a "hack" where:

```python
# just some sample
model_labels = _get_labels_from_model(nlp, "spancat")  # a flat list irrespective of spans key
for spans_key, labels in data_labels:
    for label in labels:
        if label not in model_labels:
            msg.warn("No {label} found in model labels for key {spans_key}")  # will fix the messaging here
```

I will commit an initial implementation first :)

Reply:
Updated in https://github.com/explosion/spaCy/pull/10070/commits/78c199cb5b3e6d91287adc5352dc2c953b1e042d